src/assisted_test_infra/test_infra/utils/utils.py

```diff
@@ -560,6 +561,37 @@ def get_assisted_controller_status(kubeconfig):
     return response.stdout
 
 
+def retry_success(repeat, sleep_between=30):
+    def retry_func(func):
+        def inner(*args, **kwargs):
+            start = 1
+            while start <= repeat:
+                log.info(f"retry_success decorator call func {func.__name__} {start} time")
+                res = func(*args, **kwargs)
+                if res:
+                    return res
+                else:
+                    start += 1
+                    time.sleep(sleep_between)
+            return ""
+
+        return inner
+
+    return retry_func
+
+
+@retry_success(5)
```

Comment:
don't we already use a library to perform the retry?

Reply:
The retry decorator we use in existing calls expect to get exception from caller func , the success here is a string result. we decorate function that returns a value and not exception. 
I did not find retry based on string result ... 
As long we get "" as result we repeat x times.... (the exception not raised here)